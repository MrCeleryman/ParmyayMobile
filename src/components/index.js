// @flow
import Button from "./button";

export {
	Button
};
