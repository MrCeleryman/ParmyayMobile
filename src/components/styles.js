//@flow
import { StyleSheet } from "react-native";
import theme from "../theme";

export default StyleSheet.create({
	buttonFlat: {
		backgroundColor: theme.COLOR.TRANSPARENT,
		margin: 5,
		padding: 15,
		flexDirection: "row",
		justifyContent: "center"
	},
	buttonRaised: {
		backgroundColor: theme.COLOR.ACCENT,
		margin: 5,
		flexDirection: "row",
		justifyContent: "center",
		padding: 15,
		borderRadius: 2,
		elevation: 5
	},
	buttonRaisedDisabled: {
		backgroundColor: theme.COLOR.DISABLED,
		elevation: 0
	},
	buttonFlatDisabled: {
		color: theme.COLOR.DISABLED
	},
	buttonRaisedText: {
		color: theme.COLOR.LIGHT,
		fontWeight: "500"
	},
	buttonFlatText: {
		color: theme.COLOR.ACCENT,
		fontWeight: "500"
	}
});