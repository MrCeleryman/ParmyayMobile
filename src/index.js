// @flow
import React, { Component } from "react";
import Navigation from "./nav";

export default class Parmyaymobile extends Component {
	render() {
		return (
			<Navigation />
		);
	}
}
