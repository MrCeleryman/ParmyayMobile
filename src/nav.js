// @flow
import { StackNavigator } from "react-navigation";
import Home from "./home";
import Page from "./page";

export default StackNavigator({
	Home: { screen: Home },
	Page: { screen: Page }
}, { initialRouteName: "Home" });