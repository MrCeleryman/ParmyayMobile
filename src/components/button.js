//@flow
import React from "react";
import { Text, TouchableOpacity } from "react-native";
import styles from "./styles";
import type { Button } from "../types";

export default (props: Button) => {

	const getStyle = (props: Button, style: any) => ([
		props.flat ? style.buttonFlat : style.buttonRaised,
		props.style,
		props.disabled && !props.flat ? style.buttonRaisedDisabled : null
	]);

	const getTextStyle = (props: Button, style: any) => ([
		props.flat ? style.buttonFlatText : style.buttonRaisedText,
		props.textStyle,
		props.disabled && props.flat ? style.buttonFlatDisabled : null
	]);

	return (
		<TouchableOpacity
			onPress={!props.disabled ? props.onPress : null}
			style={getStyle(props, styles)}>
			{props.text && typeof props.text == "string" ?
				<Text
					style={getTextStyle(props, styles)}>
					{props.text.toUpperCase()}
				</Text> :
				props.children}
		</TouchableOpacity>
	);
}