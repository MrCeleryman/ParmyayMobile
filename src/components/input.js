// @flow
import React from 'react';
import { TextInput, View, Text } from 'react-native';
import styles from "./styles";
import type { Input } from "../types";

export default (props: Input) => {

	const getStyle = (props: Input, style: any) => ([
		style.Input
	]);

	const getLabelStyle = (props: Input, style: any) => ([
		style.label
	]);

    const getContainerStyle = (props: Input, style: any) => ([
		style.formContainer
	]);

	return (
		<View style={getContainerStyle(props, styles)}>
            <Text style={getLabelStyle(props, styles)}>
                {props.label}
            </Text>
            <TextInput
                secureTextEntry={props.secureTextEntry}
                placeholder={props.placeholder}
                autoCorrect={false}
                value={props.value}
                onChangeText={props.onChangeText}
                style={getStyle(props, styles)}
            />
        </View>
	);
}
