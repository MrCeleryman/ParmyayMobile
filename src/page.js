// @flow
import React, { Component } from "react";
import { Text, View } from "react-native";
import styles from "./styles";

export default class Page extends Component {
	static navigationOptions = {
		title: "Test Page"
	};
	render() {
		return (
			<View style={styles.container}>
				<Text style={styles.welcome}>
					Welcome to Test Page
				</Text>
			</View>
		);
	}
}