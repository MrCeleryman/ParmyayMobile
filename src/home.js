//@flow
import React, { Component } from "react";
import { Text, TextInput, View, Animated, Image, Easing, ScrollView } from "react-native";
import { Button } from "./components";
import styles from "./styles";

export default class Home extends Component {
	spinButtonValue: number;
	spinValue: number
	spinningButton: boolean;
	spinning: boolean
	constructor(props: any) {
		super(props);
		this.spinValue = new Animated.Value(0);
		this.spinButtonValue = new Animated.Value(0);
	}

	static navigationOptions = {
		title: "ParmyayMobile"
	};

	spinButton = () => {
		if (!this.spinningButton) {
			this.spinButtonValue.setValue(0);
			this.spinningButton = true;
			Animated.timing(
				this.spinButtonValue,
				{
					toValue: 1,
					duration: 200,
					easing: Easing.linear
				}
			).start(() => this.spinningButton = false);
		}
	}

	spin = () => {
		if (!this.spinning) {
			this.spinValue.setValue(0);
			this.spinning = true;
			Animated.timing(
				this.spinValue,
				{
					toValue: 1,
					duration: 200,
					easing: Easing.linear
				}
			).start(() => this.spinning = false);
		}
	}

	navigateToPage = () => {
		this.props.navigation.navigate("Page");
	}

	render() {
		const spinButton = this.spinButtonValue.interpolate({
			inputRange: [0, 1],
			outputRange: ["0deg", "360deg"]
		});
		const spin = this.spinValue.interpolate({
			inputRange: [0, 1],
			outputRange: ["0deg", "360deg"]
		});
		return (
			<ScrollView
				style={styles.container}
				contentContainerStyle={styles.containerContent}>
				<Text style={styles.welcome}>
					Welcome to Parmyaymobile
				</Text>
				<Button text="Go to test page" onPress={this.navigateToPage} />
				<Button text="Flat button" flat />
				<Button text="Raised disabled" disabled />
				<Button text="Flat disabled" flat disabled />
				<Animated.View
					style={{ transform: [{ rotate: spinButton }] }}>
					<Button text="Spin me!" onPress={this.spinButton} />
				</Animated.View>
				<View style={styles.input}>
					<Animated.View
						style={{ transform: [{ rotate: spin }] }}>
						<Image style={styles.image} source={require("../images/chicken.jpg")} />
					</Animated.View>
					<TextInput onChangeText={this.spin} placeholder="Type here" style={styles.textInput} />
				</View>
			</ScrollView>
		);
	}
}