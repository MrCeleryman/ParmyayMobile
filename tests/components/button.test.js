import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Button } from "../../src/components";

describe("app", () => {
	it("renders correctly", () => {
		const tree = renderer.create(
			<Button />
		);
	});

	it("Snapshot matches", () => {
		expect(renderer.create(
			<Button />
		)).toMatchSnapshot();
	});
});
