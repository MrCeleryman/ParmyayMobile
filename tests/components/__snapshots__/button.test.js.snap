// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app Snapshot matches 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  isTVSelectable={true}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#5D34AF",
      "borderRadius": 2,
      "elevation": 5,
      "flexDirection": "row",
      "justifyContent": "center",
      "margin": 5,
      "opacity": 1,
      "padding": 15,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
/>
`;
